File: src/types/snapshot.ts
export interface ServiceStateRecord {
  name: string;
  startup_type: string;
  status: string;
}

export interface Snapshot {
  id: string;
  timestamp: string;
  description: string;
  restore_point_created: boolean;
  registry_export_paths: string[];
  service_states: ServiceStateRecord[];
}

/** rollback_snapshot dönüş tipi. registry_imports[i] = (path, success). */
export interface RollbackReport {
  snapshot_id: string;
  registry_imports: [string, boolean][];
  services_restored: number;
}

File: src-tauri/src/models/snapshot.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ServiceStateRecord {
    pub name: String,
    pub startup_type: String,
    pub status: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Snapshot {
    pub id: String,
    pub timestamp: DateTime<Utc>,
    pub description: String,
    pub restore_point_created: bool,
    pub registry_export_paths: Vec<String>,
    pub service_states: Vec<ServiceStateRecord>,
}
